odoo/addons/base/models/res_users.py

```diff
@@ -422,7 +422,7 @@ def _has_multiple_groups(self, group_ids):
                     SELECT 1 FROM res_groups_users_rel WHERE EXISTS(
                         SELECT r.uid
                         FROM res_groups_users_rel r
-                        WHERE r.gid IN %s""" + where_clause + """
+                        WHERE r.gid IN %s """ + where_clause + """
```

Comment:
is this necessary?

Reply:
I think so. If not, you get and invalid expression:
```
WHERE r.gid IN %sAND r.uid = %s
```

Reply:
IN requires parentheses, therefore it should not affect

Reply:
But I think that with space ir more readable

Reply:
> I think so. If not, you get and invalid expression:
> 
> ```
> WHERE r.gid IN %sAND r.uid = %s
> ```

That would mean it crashes all the time... I don't think it is the case.

Reply:
What I mean is that the space is not necessary because the `%s` parameter already comes with parentheses, so it does not fail:
 `IN (1, 8, 45)AND xxx` is a valid expression

Reply:
I think this part of the code has not been activated previously and that's why it didn't fail.

Reply:
OK, I can remove this change if you prefer, @nim-odoo

Reply:
@pedrobaeza yep, so we stick to the minimal diff. thanks

Reply:
Done, thanks